src/main/resources/avro/variantmethods.avdl

```diff
@@ -70,11 +70,11 @@ record SearchVariantsRequest {
   string variantSetId;
 
   /**
-  Only return variant calls which belong to call sets with these IDs.
+  Only return variant calls which belong to samples with these IDs.
   If an empty array, returns variants without any call objects.
   If null, returns all variant calls.
   */
-  union { null, array<string> } callSetIds = null;
+  union { null, array<string> } sampleNames = null;
```

Comment:
@lh3, your example query doesn't need to replace the callset
query, it can be an alternative.  Since I wasn't on the
conference calls, I have no idea what use cases were used to
define the queries.

My contentions is a clean conceptual data model that actually
represents the underlying data and build queries for particular
use cases will be far more satisfactory than trying to structure
the data around today's common queries.



Heng Li <notifications@github.com> writes:
> In src/main/resources/avro/variantmethods.avdl:
> 
> >    If an empty array, returns variants without any call objects.
> >    If null, returns all variant calls.
> >    */
> > -  union { null, array<string> } callSetIds = null;
> > +  union { null, array<string> } sampleNames = null;
> 
> I won't explain the technical reasons – your example is very rare in practice;
> especially in the final data release, we almost never want to put two calling
> methods in one VCF/VariantSet. I don't mind if API doesn't work elegantly in
> this extreme case (note that it still works, just not elegantly). In contrast,
> the current API model is complicating the most common use cases. It actually
> doesn't really resolve your example if you simply take VCF and SAM submissions.
> 
> —
> Reply to this email directly or view it on GitHub.*
>